File: app/src/main/java/com/emiliorg/myrestaurant/model/Plato.java
package com.emiliorg.myrestaurant.model;

import java.util.Arrays;

public class Plato {
    private int id_plato;
    private String name;
    private Ingredientes ingredientes;
    private float precio;
    private String image;

    public Plato(){

    }

    public Plato(int id_plato, String name, Ingredientes ingredients, float price) {
        this.id_plato = id_plato;
        this.name = name;
        this.ingredientes = ingredients;
        this.precio = price;

    }

    public Plato(int id_plato, String name, Ingredientes ingredients, float price, String urlPhoto) {
        this.id_plato = id_plato;
        this.name = name;
        this.ingredientes = ingredients;
        this.precio = price;

        this.image = urlPhoto;
    }

    public int getId_plato() {
        return id_plato;
    }

    public void setId_plato(int id_plato) {
        this.id_plato = id_plato;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public Ingredientes getIngredientes() {
        return ingredientes;
    }

    public void setIngredientes(Ingredientes ingredientes) {
        this.ingredientes = ingredientes;
    }

    public float getPrecio() {
        return precio;
    }

    public void setPrecio(float precio) {
        this.precio = precio;
    }

    public String getImage() {
        return image;
    }

    public void setImage(String image) {
        this.image = image;
    }



    @Override
    public String toString() {
        return "Plato{" +
                "id_plato=" + id_plato +
                ", name='" + name + '\'' +
                ", ingredientes=" + ingredientes.toString() +
                ", precio=" + precio +
                ", image='" + image + '\'' +
                '}';
    }
}

File: app/src/main/java/com/emiliorg/myrestaurant/model/Plato.kt
package com.emiliorg.myrestaurant.model

class Plato {
    private var id_plato = 0
    private var name: String? = null
    private var ingredientes: Ingredientes? = null
    private var precio = 0f
    private var image: String? = null

    constructor() {}
    constructor(id_plato: Int, name: String?, ingredients: Ingredientes?, price: Float) {
        this.id_plato = id_plato
        this.name = name
        ingredientes = ingredients
        precio = price
    }

    constructor(id_plato: Int, name: String?, ingredients: Ingredientes?, price: Float, urlPhoto: String?) {
        this.id_plato = id_plato
        this.name = name
        ingredientes = ingredients
        precio = price
        image = urlPhoto
    }

    fun getId_plato(): Int {
        return id_plato
    }

    fun setId_plato(id_plato: Int) {
        this.id_plato = id_plato
    }

    fun getName(): String? {
        return name
    }

    fun setName(name: String?) {
        this.name = name
    }

    fun getIngredientes(): Ingredientes? {
        return ingredientes
    }

    fun setIngredientes(ingredientes: Ingredientes?) {
        this.ingredientes = ingredientes
    }

    fun getPrecio(): Float {
        return precio
    }

    fun setPrecio(precio: Float) {
        this.precio = precio
    }

    fun getImage(): String? {
        return image
    }

    fun setImage(image: String?) {
        this.image = image
    }

    override fun toString(): String {
        return "Plato{" +
                "id_plato=" + id_plato +
                ", name='" + name + '\'' +
                ", ingredientes=" + ingredientes.toString() +
                ", precio=" + precio +
                ", image='" + image + '\'' +
                '}'
    }
}